File: etc/KAKAO2019 1.py
def solution(s):
    answer = 1001
    l = len(s)
    if l is 1:
        return 1

    for subStrLen in range(1, l // 2 + 1):
        startIdx = 0
        localAns = 0
        while startIdx + subStrLen <= l:
            subStr = s[startIdx:startIdx+subStrLen]
            subStartIdx = startIdx
            subLocalAns = 0
            while subStartIdx + subStrLen <= l:
                subSubStr = s[subStartIdx:subStartIdx+subStrLen]
                if subStr == subSubStr:
                    subLocalAns += 1
                    subStartIdx += subStrLen
                else:
                    break
            startIdx = subStartIdx
            if subLocalAns == 1000:
                localLen = 4
            elif subLocalAns >= 100:
                localLen = 3
            elif subLocalAns >= 10:
                localLen = 2
            elif subLocalAns > 1:
                localLen = 1
            else:
                localLen = 0
            localAns += localLen + subStrLen
        localAns += l - startIdx
        if answer > localAns:
            answer = localAns

    return answer


_s = "xababcdcdababcdcd"
print(solution(_s))

File: etc/KAKAO2019 1.cpp
#include <string>
#include <vector>
#include <iostream>
using namespace std;

int intlen(int a) {
	if (a == 1000) return 4;
	else if (a >= 100) return 3;
	else if (a >= 10) return 2;
	else if (a >= 2) return 1;
	else return 0;
}

int solution(string s) {
	int answer = 1001;
	int s_len = s.length();
	if (s_len == 1)
		return 1;

	for (int subStrLen = 1; subStrLen <= s_len / 2; subStrLen++) {
		//cout << subStrLen << endl;
		int startIdx = 0;
		int localAns = 0;
		while (startIdx + subStrLen <= s_len) {
			//cout << startIdx << "+" << subStrLen << "<=" << s_len<<endl;
			string subStr = s.substr(startIdx, subStrLen);
			int subStartIdx = startIdx;
			int subLocalAns = 0;
			while (subStartIdx + subStrLen <= s_len) {
				//cout << subStartIdx << "+" << subStrLen << "<=" << s_len << endl;
				string cmp = s.substr(subStartIdx, subStrLen);
				if (subStr.compare(cmp) == 0) {
					subLocalAns++;
					subStartIdx += subStrLen;
				}
				else {
					break;
				}
			}
			startIdx = subStartIdx;
			localAns += intlen(subLocalAns) + subStrLen;
		}
		localAns += s_len - startIdx;
		answer = answer > localAns ? localAns : answer;
	}
	return answer;
}

int main() {
	string s = "abcabcabcabcdededededede";
	cout << solution(s) << endl;

	return 0;
}